app/domain/authentication/authn_oidc/v2/client.rb

```diff
@@ -104,7 +104,7 @@ def discovery_information(invalidate: false)
             skip_nil: true
           ) do
             @discovery_configuration.discover!(@authenticator.provider_uri)
-          rescue HTTPClient::ConnectTimeoutError, Errno::ETIMEDOUT => e
+          rescue Errno::ETIMEDOUT => e
```

Comment:
I was trying to figure out whether we should replace these `HTTPClient` errors with `Faraday` ones, but I'm not sure that it's necessary. The error returned by `discover!` can only be type [`OpenIDConnect::Discovery::DiscoveryFailed`](https://github.com/nov/openid_connect/blob/master/lib/openid_connect/discovery.rb#L4C19-L4C19), which is too non-specific, I think.

Reply:
Yep it's the same exception type for basically any network error that occurs. I suppose it would be doable to parse it out of the error message but that seemed a bit hacky and unnecessary